sql/src/main/java/io/crate/planner/consumer/QueryAndFetchConsumer.java

```diff
@@ -376,27 +359,4 @@ private static void tryApplySizeHint(@Nullable Integer requiredPageSize, Limits
         }
         return Lists2.concatUnique(qsOutputs, orderBy.orderBySymbols());
     }
-
-    private static final  class NoPredicateVisitor extends SymbolVisitor<Void, Void> {
-
-        private static final NoPredicateVisitor NO_PREDICATE_VISITOR = new NoPredicateVisitor();
-
-        private NoPredicateVisitor() {
-        }
-
-        static void ensureNoMatchPredicate(Symbol symbolTree) {
-            NO_PREDICATE_VISITOR.process(symbolTree, null);
-        }
-
-        @Override
-        public Void visitFunction(Function symbol, Void context) {
-            if (symbol.info().ident().name().equals(MatchPredicate.NAME)) {
```

Comment:
I wonder if `match` is the only operation? I guess Lucene type of operations don't work in general.

Reply:
Currently `match` is the only lucene specific predicate